#!/usr/bin/env python3

PACKAGE = "object_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# ROI cfg parameters
gen.add("xMin", double_t, 0, "xMinValue", -1.2, -200, 10) #-0.55
gen.add("xMax", double_t, 0, "xMaxValue", 6.0, -10, 200)
gen.add("yMin", double_t, 0, "yMinValue", -1.6, -200, 10)#원래 -3이였음
gen.add("yMax", double_t, 0, "yMaxValue", 1.6, -10, 200)#원래3이였음
gen.add("zMin", double_t, 0, "zMinValue", -0.42, -2, 0) #원래 -0.42였음
gen.add("zMax", double_t, 0, "zMaxValue", -0.2, -2, 100)#원래 -0.2였음

# DBScan cfg parameters
gen.add("minPoints", int_t, 0, "minPointsValue", 10, 1, 100)
gen.add("epsilon", double_t, 0, "epsilonValue", 0.4, 0, 2)
gen.add("minClusterSize", double_t, 0, "minClusterSizeValue", 10, 0, 100)
gen.add("maxClusterSize", double_t, 0, "maxClusterSizeValue", 2200, 0, 10000)

# VoxelGrid parameter
gen.add("leafSize", double_t, 0, "leafSizeValue", 0.03, 0, 1)

exit(gen.generate(PACKAGE, "object_detector_node", "objectDetector"))
